File: DotNet/Normalization/Application/Models/Operations/Business/OperationModel.cs
using System.Diagnostics.CodeAnalysis;

namespace LantanaGroup.Link.Normalization.Application.Models.Operations.Business
{
    [ExcludeFromCodeCoverage]
    public class OperationModel
    {
        public Guid Id { get; set; }
        public string FacilityId { get; set; } = string.Empty;
        public string OperationJson { get; set; } = string.Empty;
        public string OperationType { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public bool IsDisabled { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime? ModifyDate { get; set; }
        public List<VendorVersionOperationPresetModel> VendorPresets { get; set; } = new List<VendorVersionOperationPresetModel>();
        public List<OperationResourceTypeModel> OperationResourceTypes { get; set; } = new();
    }
}

File: Web/Admin.UI/src/app/components/tenant/global-operations/models/operation-model.ts
import { CodeMapOperation } from "src/app/interfaces/normalization/code-map-operation-interface";
import { ConditionalTransformOperation } from "src/app/interfaces/normalization/conditional-transformation-operation-interface";
import { CopyPropertyOperation } from "src/app/interfaces/normalization/copy-property-interface";
import { IOperation } from "src/app/interfaces/normalization/operation.interface";
import { PaginationMetadata } from "src/app/models/pagination-metadata.model";

export interface OperationModel {
  id: string;
  facilityId: string;
  operationJson: string;
  parsedOperationJson: CopyPropertyOperation | ConditionalTransformOperation | CodeMapOperation | IOperation;
  operationType: string;
  description: string;
  isDisabled: boolean;
  createDate: string;
  modifyDate?: string;
  resources: ResourceModel[];
  vendorPresets: VendorOperationPresetModel[];
}

export interface ResourceModel {
  resourceTypeId: string;
  resourceName: string;
}

export interface VendorOperationPresetModel {
  id: string;
  vendor?: string;
  versions?: string;
  description?: string;
  createDate: string; 
  modifyDate?: string;
}

export interface IPagedOperationModel {
  records: OperationModel[];
  metadata: PaginationMetadata;
}